Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -2581,6 +2588,274 @@ def pin_message():
         return_error(f"{slack_error}")
 
 
+def list_channels():
+    """
+    List the conversations in the workspace
+    """
+    channel_types = demisto.args().get('channel_types')
+    # Default for the SDK is public channels, but users can specify "public_channel", "private_channel", "mpim", and "im"
+    # Multiple values can be passed for this argument as a comma separated list
+    if channel_types is None:
+        channel_types = 'public_channel'
+    # By default archived channels are NOT included by the SDK. Explicitly set this if not set from the CLI or set to False
+    exclude_archived = demisto.args().get('exclude_archived')
+    if exclude_archived is None:
+        exclude_archived = 'true'
+
+    limit = demisto.args().get('limit')
+    if limit is None:
+        limit = 100
+
+    body = {
+        'types': channel_types,
+        'exclude_archived': exclude_archived,
+        'limit': limit
+    }
+
+    cursor = demisto.args().get('cursor')
+    if cursor != None:
+        body['cursor'] = cursor
+
+    raw_response = send_slack_request_sync(CLIENT, 'conversations.list', http_verb="GET", body=body)
+    # Provide an option to only select a channel by a name. Instead of returning a full list of results this allows granularity
+    # Supports a single channel name
+    name_filter = demisto.args().get('name_filter')
+
+    if name_filter != None:
+        channels = None
+        for channel in raw_response['channels']:
+            if channel['name'] == name_filter:
+                channels = [channel]
+                break
+            else:
+                continue
+        if channels == None:
+            return_error(f"No channel found with name: {name_filter}")
+    else:
+        channels = raw_response['channels']
+    # Force list for consistent parsing
+    if type(channels) == dict:
+        channels = [channels]
+    context = []
+    for channel in channels:
+        creator = channel['creator']
+        body = {
+            'user':creator
+        }
+        creator_details_response = send_slack_request_sync(CLIENT, 'users.info', http_verb="GET", body=body)
```

Comment:
get_user_by_name does not have the ability to search by Slack User ID. It can search by Name, Real Name, and Email. I don't mind updating the function to search by User ID, but I am requesting guidance on how to update this function to search by User ID. 

We could also add rate limiting to this part of the code, however I think this is not as valuable as a fix as updating the get_user_by_name function.

Reply:
In that case, then let's leave it as is. I'm hesitant to make a change to the function since it could have unintended side effects elsewhere.